runtime/src/tiered_storage/hot.rs

```diff
@@ -123,15 +131,77 @@ impl TieredAccountMeta for HotAccountMeta {
     fn supports_shared_account_block() -> bool {
         false
     }
+
+    /// Returns the epoch that this account will next owe rent by parsing
+    /// the specified account block.  None will be returned if this account
+    /// does not persist this optional field.
+    fn rent_epoch(&self, data_block: &[u8]) -> Option<Epoch> {
+        let offset = self.optional_fields_offset(data_block)
+            + AccountMetaOptionalFields::rent_epoch_offset(self.flags());
+        self.flags()
+            .has_rent_epoch()
+            .then(|| byte_block::read_type::<Epoch>(data_block, offset).copied())
+            .flatten()
+    }
+
+    /// Returns the account hash by parsing the specified account block.  None
+    /// will be returned if this account does not persist this optional field.
+    fn account_hash<'a>(&self, data_block: &'a [u8]) -> Option<&'a Hash> {
+        let offset = self.optional_fields_offset(data_block)
+            + AccountMetaOptionalFields::account_hash_offset(self.flags());
+        self.flags()
+            .has_account_hash()
+            .then(|| byte_block::read_type::<Hash>(data_block, offset))
+            .flatten()
+    }
+
+    /// Returns the write version by parsing the specified account block.  None
+    /// will be returned if this account does not persist this optional field.
+    fn write_version(&self, data_block: &[u8]) -> Option<StoredMetaWriteVersion> {
+        let offset = self.optional_fields_offset(data_block)
+            + AccountMetaOptionalFields::write_version_offset(self.flags());
+        self.flags
+            .has_write_version()
+            .then(|| byte_block::read_type::<StoredMetaWriteVersion>(data_block, offset).copied())
+            .flatten()
+    }
+
+    /// Returns the offset of the optional fields based on the specified account
+    /// block.
+    fn optional_fields_offset(&self, data_block: &[u8]) -> usize {
+        data_block
+            .len()
+            .saturating_sub(AccountMetaOptionalFields::size_from_flags(&self.flags))
+    }
+
+    /// Returns the length of the data associated to this account based on the
+    /// specified account block.
+    fn data_len(&self, data_block: &[u8]) -> usize {
+        self.optional_fields_offset(data_block)
+            .saturating_sub(self.account_data_padding() as usize)
+    }
```

Comment:
> We use data size up above in with_data_size, can this replace len with size?

Yes, and they represent the same thing.  Will make them use the same term.

> We are overloading data, it seems. 

The "data" this file refers to is "account data" so data and account data are the same thing in this file.

 Given this, I think it's better to use the same term.
 
 Since this file is already about HotAccountMeta, I think simply calling it "data" should be enough.
 If you think account data isn't too lengthy, we can also call it account data to avoid confusion.
 
 > data_size_for_shared_account_block() -> shared_account_block_data_size() or shared_account_block_size() (if the data part is not necessary)
 
 This function is a bit confusing.  Let me remove it for now and see whether I can come up with something cleaner.

Reply:
Created #32317 for removing the confusing API.  Haven't renamed `data_len()` to `data_size()`.  Will have all the renamings in a separate PR as this will also update `meta.rs`.